scripts/kmod_build.py

```diff
@@ -68,7 +68,7 @@ def build_module(output_dir, module_ko, kdir, module_dir, make_command, make_arg
     # Sanitize the environment - we should only use build options passed in via
     # the command line.
     env = dict(os.environ)
-    for var in ["ARCH", "CROSS_COMPILE", "CC", "HOSTCC", "CLANG_TRIPLE", "KBUILD_EXTRA_SYMBOLS"]:
+    for var in ["ARCH", "CROSS_COMPILE", "CC", "HOSTCC", "CLANG_TRIPLE", "KBUILD_EXTRA_SYMBOLS", "LD"]:
```

Comment:
The kernel make doesn't read LD from an environment variable (it's assigned with '='), so we can drop this. Actually, this is the same for CC and HOSTCC

Reply:
In the main `Makefile` I think you're correct. But there's stuff like this in various other sub-make files and helper scripts:

- `tools/lib/subcmd/Makefile:12` has `CC ?= $(CROSS_COMPILE)gcc`
- `tools/perf/util/setup.py:8` has `cc = getenv("CC")
- `tools/build/Makefile` and `tools/objtool/Makefile` both have `HOSTCC ?= gcc`

I haven't looked into exactly how these are invoked, but wouldn't it be safer just to remove potential conflicting values from the environment just in case?

Reply:
Hmm. That's annoying. Let's commit as is then.